### PR TITLE
Users/search_support

### PR DESCRIPTION
**Pull request by sahils-tenable**
- Added support for search endpoint.
- This PR depends upon #69

### CODE REVIEW

**Review by harshshah-tenable — APPROVED** [review]

File: tenable/io/v3/users/api.py [hunk 1]
```diff
@@ -230,171 +253,253 @@ def two_factor(
...
-            filter -- tuple
-                ('field_name', 'operator', 'value') --
-                ('and', ('test', 'oper', '1'),
-                ('test', 'oper', '2'))
-            sort -- list of dictionary
-                    [{'property': 'field_name', 'order': 'asc'}]
-                 -- sort is not supported by search api for now.
-            limit -- integer = (10)
-            next -- integer = (10)
+        Retrieves the users data.
```
Comment (ngaglani-tenable) [inline tenable/io/v3/users/api.py]: Update the description as per the latest one.

File: tests/io/v3/users/test_users_api.py [hunk 2]
```diff
@@ -34,19 +38,103 @@
...
+
+    iterator = api.v3.users.search(
+        fields=fields, limit=200, sort=sort, filter=filters
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    event_list = []
+    for event in iterator:
+        event_list.append(event)
+    assert len(event_list) == api_response['pagination']['total']
```
Comment (alpesh-te) [inline tests/io/v3/users/test_users_api.py]: Rewrite it to get rid of loop.

File: tenable/io/v3/users/api.py [hunk 3]
```diff
@@ -230,171 +253,253 @@ def two_factor(
...
+                specific page of data.
+            return_csv (bool, optional):
+                If set to true, it will return the CSV response or
+                iterable (based on return_resp flag). Iterator returns all
+                rows in text/csv format for each call with row headers.
+        Returns:
+            Iterable:
+                The iterable that handles the pagination for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (alpesh-te) [inline tenable/io/v3/users/api.py]: Do string update

**Review by alpesh-te — CHANGES_REQUESTED** [review]
Minor fixes, other than that LGMT

File: tenable/io/v3/users/api.py (hunk 3 above)
Comment (alpesh-te) [reply tenable/io/v3/users/api.py]: Addressed

File: tests/io/v3/users/test_users_api.py (hunk 2 above)
Comment (alpesh-te) [reply tests/io/v3/users/test_users_api.py]: Addressed

File: tenable/io/v3/users/api.py (hunk 1 above)
Comment (alpesh-te) [reply tenable/io/v3/users/api.py]: Addressed

**Review by alpesh-te — APPROVED** [review]
LGMT

**Pull request merged**